src/Form/AppAnalyticsFormBase.php

```diff
@@ -445,14 +445,16 @@ protected function generateResponse(array &$form, AppInterface $app, string $met
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $url = $this->urlGenerator->generateFromRoute('<current>', [], [
+    $route = \Drupal::routeMatch();
```

Comment:
Pls dependency inject the route matcher service.

Reply:
I wasn't sure if we need to inject services when using them for one off situations like this, but I do agree with @mxr576 after looking at [coding standards](https://www.drupal.org/docs/8/api/services-and-dependency-injection/services-and-dependency-injection-in-drupal-8#s-using-dependency-injection)

> Dependency injection is the preferred method for accessing and using services in Drupal 8 and should be used whenever possible.

Reply:
This should not be needed  - see https://github.com/apigee/apigee-edge-drupal/pull/266#issuecomment-534248353.